tests/Doctrine/Tests/KeyValueStore/Storage/RiakStorageTest.php

```diff
@@ -67,182 +78,102 @@ public function testRequiresCompositePrimaryKeys()
 
-        $bucket->expects($this->once())
-             ->method('get')
-             ->will($this->returnValue($objectMock));
+        $namespace = new RiakNamespace('default', 'riak-test');
+        $location = new RiakLocation($namespace, 'foobar');
 
-        $that = $this;
-        $objectMock->expects($this->once())
-            ->method('setData')
-            ->will($this->returnCallback(function ($data) use ($that) {
-                $that->assertEquals(['title' => 'Riak cookbook'], $data);
-            }));
+        $fetch = FetchValue::builder($location)->build();
 
-        $objectMock->expects($this->once())
-            ->method('store');
+        $json = (string) $this->client
+            ->execute($fetch)
+            ->getValue()
+            ->getValue();
 
-        $this->storage->update('riak-test', 'foobar', ['title' => 'Riak cookbook']);
+        $this->assertSame($data, json_decode($json, true));
     }
 
-    public function testDelete()
+    /**
+     * @depends testInsert
+     */
+    public function testUpdate()
     {
-        $objectMock = $this->getMockBuilder('Riak\Object')
-            ->disableOriginalConstructor()
-            ->getMock();
+        $data = [
+            'title' => 'Riak update',
+        ];
 
-        $bucket = $this->getMockBuilder('Riak\Bucket')
-            ->disableOriginalConstructor()
-            ->getMock();
+        $this->storage->insert('riak-test', 'foobar', [
+            'title' => 'Riak insert',
+        ]);
 
-        $this->riak->expects($this->once())
-            ->method('bucket')
-            ->will($this->returnValue($bucket));
+        $this->storage->insert('riak-test', 'foobar', $data);
 
-        $bucket->expects($this->once())
-            ->method('get')
-            ->with('foobar')
-            ->will($this->returnValue($objectMock));
+        $namespace = new RiakNamespace('default', 'riak-test');
+        $location = new RiakLocation($namespace, 'foobar');
 
-        $objectMock->expects($this->once())
-            ->method('exists')
-            ->will($this->returnValue(true));
+        $fetch = FetchValue::builder($location)->build();
 
-        $objectMock->expects($this->once())
-            ->method('delete');
+        $json = (string) $this->client
+            ->execute($fetch)
+            ->getValue()
+            ->getValue();
 
-        $this->storage->delete('riak-test', 'foobar');
+        $this->assertSame($data, json_decode($json, true));
     }
 
-    public function testDeleteWithNotExistKey()
+    /**
+     * @depends testInsert
+     */
+    public function testDelete()
     {
-        $objectMock = $this->getMockBuilder('Riak\Object')
-            ->disableOriginalConstructor()
-            ->getMock();
-
-        $bucket = $this->getMockBuilder('Riak\Bucket')
-            ->disableOriginalConstructor()
-            ->getMock();
+        $this->storage->insert('riak-test', 'foobar', [
```

Comment:
What happens if the insert wasn't successful here? Should we check for an inserted value here before we can be sure that `delete` works despite that this test has a dependency to `testInsert()`?

Reply:
I've reused `testInsert` inside `testDelete` so the insertion is checked in deletion too